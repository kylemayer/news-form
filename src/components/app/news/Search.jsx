/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ searchWord, onSubmit, onSearchChange }) => (
  <form onSubmit={onSubmit}>
    <label htmlFor="searchWord">Search</label>
    <input
      id="searchWord"
      type="text"
      value={searchWord}
      onChange={onSearchChange}
    />
    <button aria-label="find-articles">Submit</button>
  </form>
);

Search.protoTypes = {
  searchWord: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onSearchChange: PropTypes.func.isRequired,
};

export default Search;
