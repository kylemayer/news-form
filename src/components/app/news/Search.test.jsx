import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Search from './Search';

describe('Search file', () => {
  it('should render an input form and a button', async () => {
    render(<Search />);

    const input = await screen.findByLabelText('search');
    userEvent.type(input, 'Bidden');

    const submitButton = await screen.findByRole('button', {
      name: 'find-articles',
    });

    expect(input).toMatchSnapshot();
    expect(submitButton).toMatchSnapshot();
  });
});
