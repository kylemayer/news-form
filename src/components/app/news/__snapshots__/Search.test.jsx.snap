// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search file should render an input form and a button 1`] = `
<input
  aria-label="search"
  type="text"
  value=""
/>
`;

exports[`Search file should render an input form and a button 2`] = `
<button
  aria-label="find-articles"
>
  Submit
</button>
`;
